pkg/util/label.go

```diff
@@ -42,6 +42,8 @@ func RetainLabels(desired *unstructured.Unstructured, observed *unstructured.Uns
 		}
 		labels[key] = value
 	}
+	// Delete following one line in release-1.9
+	delete(labels, workv1alpha2.WorkUIDLabel)
```

Comment:
Please @chaunceyjiang take a look if this(edit common utils) is acceptable,

Reply:
This line of code is redundant.

Reply:
I didn't get it. If without this line, the old label will exist in the member clusters

Reply:
https://github.com/karmada-io/karmada/pull/4199/files#diff-fe0321ed56bffdf459be14d2debdeb10c065aa28bf5dc4e092142d6b4bccbe4bR146

Because we have already removed this label from the control plane.

Refer to:
https://github.com/karmada-io/karmada/pull/3099

Reply:
Let me try

Reply:
The label will be deleted if the label key(`work.karmada.io/uid`) in annotation `resourcetemplate.karmada.io/managed-labels` of member clusters' resourcetemplate. But it doesn't exist because it's added when creating resourcetemplate in member clusters: https://github.com/karmada-io/karmada/blob/5180ad50342978a0215a287a7514f23a6ca8e470/pkg/controllers/execution/execution_controller.go#L193

It should insert this label here(Forgot to make changes in the previous version - 1.7.0):
https://github.com/karmada-io/karmada/blob/5180ad50342978a0215a287a7514f23a6ca8e470/pkg/util/label.go#L107

Reply:
So, we have to keep this for this version.

Reply:
Got it.